CI: PyPy3 tests on Azure failed for a couple recent PRs

Both https://github.com/numpy/numpy/pull/16083 and https://github.com/numpy/numpy/pull/14924 show this error in the Azure Linux PyPy3 error log:

```
=================================== FAILURES ===================================
_________________________ TestCApiAccess.test_getitem __________________________

self = <numpy.core.tests.test_indexing.TestCApiAccess object at 0x0000000009dfd2b8>

    def test_getitem(self):
        subscript = functools.partial(array_indexing, 0)
    
        # 0-d arrays don't work:
        assert_raises(IndexError, subscript, np.ones(()), 0)
        # Out of bound values:
        assert_raises(IndexError, subscript, np.ones(10), 11)
>       assert_raises(IndexError, subscript, np.ones(10), -11)

self       = <numpy.core.tests.test_indexing.TestCApiAccess object at 0x0000000009dfd2b8>
subscript  = functools.partial(<builtin_function_or_method object at 0x00007fc4341d21e0>, 0)

numpy/core/tests/test_indexing.py:1268: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
../../../pypy3/lib-python/3/unittest/case.py:733: in assertRaises
    return context.handle('assertRaises', args, kwargs)
        args       = (functools.partial(<builtin_function_or_method object at 0x00007fc4341d21e0>, 0), array([1., 1., 1., 1., 1., 1., 1., 1., 1., 1.]), -11)
        context    = None
        expected_exception = <class 'IndexError'>
        kwargs     = {}
        self       = <numpy.testing._private.utils._Dummy testMethod=nop>
../../../pypy3/lib-python/3/unittest/case.py:178: in handle
self       = <unittest.case._AssertRaisesContext object at 0x0000000009dfd280>
standardMsg = 'IndexError not raised by functools.partial(<builtin_function_or_method object at 0x00007fc4341d21e0>, 0)'

../../../pypy3/lib-python/3/unittest/case.py:135: AssertionError

```

--- a/tools/pypy-test.sh
+++ b/tools/pypy-test.sh
@@ -6,8 +6,8 @@ set -o pipefail
 # Print expanded commands
 set -x
 
-sudo apt-get -yq update
-sudo apt-get -yq install libatlas-base-dev liblapack-dev gfortran-5 python3-urllib3
+#sudo apt-get -yq update
+#sudo apt-get -yq install libatlas-base-dev liblapack-dev gfortran-5 python3-urllib3
 F77=gfortran-5 F90=gfortran-5 \
 
 # Download the proper OpenBLAS x64 precompiled library
@@ -27,8 +27,8 @@ include_dirs = $target/lib:$LIB
 runtime_library_dirs = $target/lib
 EOF
 
-echo getting PyPy 3.6 nightly
-wget -q http://buildbot.pypy.org/nightly/py3.6/pypy-c-jit-latest-linux64.tar.bz2 -O pypy.tar.bz2
+echo getting PyPy 3.6-v7.3.1
+wget -q https://downloads.python.org/pypy/pypy3.6-v7.3.1-linux64.tar.bz2 -O pypy.tar.bz2
 mkdir -p pypy3
 (cd pypy3; tar --strip-components=1 -xf ../pypy.tar.bz2)
 pypy3/bin/pypy3 -mensurepip

